### PR TITLE
service: Log query runtime errors

### PR DESCRIPTION
**Pull request by mattnibs**
Part of #4215

### CODE REVIEW

File: service/handlers.go [hunk 1]
```diff
@@ -88,19 +88,22 @@ func handleQuery(c *Core, w *ResponseWriter, r *Request) {
 		select {
 		case <-timer.C:
 			if err := writer.WriteProgress(meter.Progress()); err != nil {
+				w.Logger.Warn("Error Writing Progress", zap.Error(err))
```
Comment (nwt) [inline service/handlers.go:91]: Capitalize just the first word.
```suggestion
				w.Logger.Warn("Error writing progress", zap.Error(err))
```

File: service/ztests/query-runtime-error-log.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+script: |
+  LAKE_EXTRA_FLAGS='-log.path=service.log' source service.sh
+  zed create -use -q test
+  echo '{ts:1,union:"hi"((int64,string))}' | zed load -q -
+  curl -o /dev/null -H 'Accept: application/x-parquet' -d '{"query":"from test"}' $ZED_LAKE/query 
+  zq -Z 'cut level, msg, error' service.log
+
+
```
Comment (nwt) [inline service/ztests/query-runtime-error-log.yaml:8]:
```suggestion

```

File: service/ztests/query-runtime-error-log.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+script: |
+  LAKE_EXTRA_FLAGS='-log.path=service.log' source service.sh
+  zed create -use -q test
+  echo '{ts:1,union:"hi"((int64,string))}' | zed load -q -
```
Comment (nwt) [inline service/ztests/query-runtime-error-log.yaml:4]: Nit: You can use a simpler input here (though it'll give you a different error).
```suggestion
  echo {} | zed load -q -
```

Comment (mattnibs) [reply service/ztests/query-runtime-error-log.yaml:4]: Fine by me as long as we get the error.

**Pull request merged**